### ISSUE TITLE
Attach/Detach Open Cover with .Net running Application for code coverage.

### ISSUE THREAD

**Issue by hemantattri** [OpenCover/opencover#648]
Please provide the following information when submitting an issue, where appropriate replace the `[ ]` with a `[X]`

### My Framework

- [ ] .NET 2
- [ ] .NET 3.5
- [X] .NET 4
- [ ] .NET 4.5
- [ ] .NET 4.6
- [ ] .NET 4.6.1
- [ ] .NET 4.6.2
- [ ] .NET Core 1.0.0
- [ ] Something else

### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL)
- [ ] Windows 8
- [ ] Windows 8.1
- [ ] Windows 10
- [ ] Windows 10 IoT Core
- [ ] Windows Server 2012
- [ ] Windows Server 2012 R2
- [ ] Windows Server 2016

### I have already...

- [ ] repeated the problem using the latest stable release of OpenCover.
- [ ] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf).
- [ ] have looked at the opencover output xml file in an attempt to resolve the issue.
- [ ] reviewed the [current issues](https://github.com/opencover/opencover/issues) to check that the issue isn't already known.

### My issue is related to (check only those which apply):

- [ ] no coverage being recorded
- [ ] 32 or 64 bit support
- [X] feature request

### Expected Behavior

I want to get the coverage data for an application which is running independently.
like when I want the coverage data then can attach Open Cover with running application and when say stop then it should generate the coverage XML.

To Sum Up.
I don't want to start the application with Open Cover and then Stop for getting the coverage XML.
Can we do it just start coverage and then do some action(i.e. not stopping the application) to get the XML from open cover

### Actual Behavior

No options

### Steps to reproduce the problem:

Fill me in...

**Comment by sawilde** [comment]
Sorry but this is not possible with OpenCover.

Attach and Detach is possible with a profiler, but OpenCover uses certain functionality that isn't available to an Attached profiler - see the section [Specific Callback Limitations](https://blogs.msdn.microsoft.com/davbr/2009/11/04/clr-v4-profiler-attach-basics-with-sample-code/)

**Comment by hemantattri** [comment]
Hi - Thanks for your response, will look into this and update you.

**Comment by hemantattri** [comment]
can you please tell me in which language OpenCover is written, I think it is in c++.

**Comment by sawilde** [comment]
the profiler is written in c++, the host in c#

**Comment by jmecosta** [comment]
@sawilde we want similar functionality, basically because now SonarQube decided to merge all coverage into a single coverage number we no longer have the possibility to display system coverage vs unit test coverage. however the tests created to test ui are relevant, however running opencover from start will wrongly increase the coverage for the all project.

i supose the way to overcome, is to tell OpenCover to filter everything else except of the class under test? do you have a better suggestion?

**Comment by sawilde** [comment]
@jmecosta  not really, I avoid using coverage for anything other than unit tests so not a problem I have needed to solve.

**Comment by sawilde** [comment]
Closing this issue as the initial request is not possible with the current state of the profiler API.
